src/main/java/com/example/demo/productComponent/helper/validator/ProductValidator.java

```diff
@@ -0,0 +1,96 @@
+package com.example.demo.productComponent.helper.validator;
+
+import com.example.demo.productComponent.api.dtos.PatchProductDTO;
+import com.example.demo.productComponent.api.dtos.PostProductDTO;
+import com.example.demo.productComponent.repository.ProductCompatibleModifierRepository;
+import com.example.demo.productComponent.repository.ProductModifierRepository;
+import com.example.demo.productComponent.repository.ProductRepository;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.UUID;
+
+@AllArgsConstructor
+@Component
+public class ProductValidator {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProductValidator.class);
+
+    private final ProductRepository productRepository;
+    private final ProductModifierRepository productModifierRepository;
+    private final ProductCompatibleModifierRepository productCompatibleModifierRepository;
+
+     public boolean productsExist(List<UUID> productIds) {
+         long count = productRepository.countByIdIn(productIds);
+         return count == productIds.size();
+     }
+
+     public boolean modifiersExist(List<UUID> modifierIds) {
+         long count = productModifierRepository.countByIdIn(modifierIds);
+         return count == modifierIds.size();
+     }
+
+    public void productExists(UUID productId) {
+        if (!productRepository.existsById(productId)) {
+            logger.error("Product with id {} does not exist", productId);
+            throw new IllegalArgumentException("Product with id " + productId + " does not exist");
```

Comment:
Illegal argument man rodos nelabai tinka jei objektas neegzistuoja, gal sukurk koki savo exceptiona?

Reply:
kursim globalias klases del to, tai veliau noriu exceptionus taisyt, cia tik placeholderis